src/serac/physics/utilities/finite_element_state.cpp

```diff
@@ -6,39 +6,41 @@
 
 #include "serac/physics/utilities/finite_element_state.hpp"
 
-#include "serac/infrastructure/logger.hpp"
-
 namespace serac {
 
+// Definition of extern variable
+QuadratureData<void> dummy_qdata;
+
 FiniteElementState::FiniteElementState(mfem::ParMesh& mesh, FiniteElementState::Options&& options)
     : mesh_(mesh),
       coll_(options.coll ? std::move(options.coll)
                          : std::make_unique<mfem::H1_FECollection>(options.order, mesh.Dimension())),
-      space_(
-          std::make_unique<mfem::ParFiniteElementSpace>(&mesh, &retrieve(coll_), options.vector_dim, options.ordering)),
+      space_(std::make_unique<mfem::ParFiniteElementSpace>(&mesh, &detail::retrieve(coll_), options.vector_dim,
+                                                           options.ordering)),
       // When left unallocated, the allocation can happen inside the datastore
       // Use a raw pointer here when unallocated, lifetime will be managed by the DataCollection
-      gf_(options.alloc_gf
-              ? MaybeOwningPointer<mfem::ParGridFunction>{std::make_unique<mfem::ParGridFunction>(&retrieve(space_))}
-              : MaybeOwningPointer<mfem::ParGridFunction>{new mfem::ParGridFunction(&retrieve(space_),
-                                                                                    static_cast<double*>(nullptr))}),
-      true_vec_(&retrieve(space_)),
+      gf_(options.alloc_gf ? detail::MaybeOwningPointer<mfem::ParGridFunction>{std::make_unique<mfem::ParGridFunction>(
+                                 &detail::retrieve(space_))}
+                           : detail::MaybeOwningPointer<mfem::ParGridFunction>{new mfem::ParGridFunction(
+                                 &detail::retrieve(space_), static_cast<double*>(nullptr))}),
```

Comment:
These long single-line expressions are hard for me to read, is it practical to put them in a separate function?